nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -817,4 +823,40 @@ Costmap2DROS::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameter
   return result;
 }
 
+void Costmap2DROS::getCostCallback(
+  const std::shared_ptr<rmw_request_id_t>,
+  const std::shared_ptr<nav2_msgs::srv::GetCost::Request> request,
+  const std::shared_ptr<nav2_msgs::srv::GetCost::Response> response)
+{
+  unsigned int mx, my;
+
+  auto costmap = layered_costmap_->getCostmap();
+
+  if (request->use_footprint) {
```

Comment:
Sorry, what I meant here is to use the footprint for collision checking. Given a pose `(x, y, theta)` determine if its in collision with the environment. You should be able to use this object: https://github.com/ros-navigation/navigation2/blob/f73fa2499b54bfb346fa3dcdd7a5a200ba349752/nav2_costmap_2d/src/footprint_collision_checker.cpp#L35